addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/UniFiController.java

```diff
@@ -0,0 +1,403 @@
+                if (siteInsightsMap != null) {
+                    insightsMap.putAll(siteInsightsMap);
+                }
+            }
+        }
+        return insightsMap;
+    }
+
+    private boolean belongsToSite(UniFiClient client, String siteName) {
+        boolean result = true; // mgb: assume true = proof by contradiction
+        if (StringUtils.isNotEmpty(siteName)) {
+            UniFiSite s = sitesCache.get(client.getSiteId());
+            // mgb: if the 'site' can't be found or the name doesn't match...
+            if (s == null || !s.matchesName(siteName)) {
+                // mgb: ... then the client doesn't belong to this thing's configured 'site' and we 'filter' it
+                result = false;
+            }
+        }
+        return result;
+    }
+
+    // Public API functions
+
+    public synchronized void refresh() throws UniFiException {
+        sitesCache = getSites();
+        devicesCache = getDevices();
+        clientsCache = getClients();
+        insightsCache = getInsights();
+    }
+
+    public synchronized UniFiWiredClient getWiredClient(String mac, String site) {
+        UniFiClient client = clientsCache.get(mac);
+
+        // mgb: short circuit
+        if (client == null || BooleanUtils.isNotTrue(client.isWired()) || !belongsToSite(client, site)) {
+            return null;
+        }
+
+        // mgb: instanceof check just for type / cast safety
+        return (client instanceof UniFiWiredClient ? (UniFiWiredClient) client : null);
+    }
+
+    public synchronized UniFiWirelessClient getWirelessClient(String mac, String site) {
+        // mgb: first check active clients and fallback to insights if not found
+        UniFiClient client = clientsCache.containsKey(mac) ? clientsCache.get(mac) : insightsCache.get(mac);
+
+        // mgb: short circuit
+        if (client == null || BooleanUtils.isNotTrue(client.isWireless()) || !belongsToSite(client, site)) {
+            return null;
+        }
+
+        // mgb: instanceof check just for type / cast safety
+        return (client instanceof UniFiWirelessClient ? (UniFiWirelessClient) client : null);
+    }
+
+    public boolean isWirelessClientOnline(UniFiWirelessClient client) {
+        boolean online = false;
+        if (client != null) {
+            Calendar lastSeen = client.getLastSeen();
+            if (lastSeen == null) {
+                logger.warn("Could not determine if client is online: mac = {}, lastSeen = null", client.getMac());
+            } else {
+                Calendar considerHome = (Calendar) lastSeen.clone();
+                considerHome.add(Calendar.SECOND, config.getConsiderHome().intValue());
+                Calendar now = Calendar.getInstance();
+                online = (now.compareTo(considerHome) < 0);
+            }
+        }
+        return online;
+    }
+
+    public void logout() {
+        try {
+            get(getLogoutUrl());
+        } catch (Exception e) {
+            // nop
```

Comment:
Logging and narrowing down the caught exception to `IOException | URISyntaxException`  would be better.

Reply:
> Logging and narrowing down the caught exception to IOException | URISyntaxException would be better.